lms/djangoapps/instructor_task/models.py

```diff
@@ -192,11 +188,28 @@ def from_config(cls, config_name):
         Return one of the ReportStore subclasses depending on django
         configuration. Look at subclasses for expected configuration.
         """
-        storage_type = getattr(settings, config_name).get("STORAGE_TYPE")
-        if storage_type.lower() == "s3":
-            return S3ReportStore.from_config(config_name)
-        elif storage_type.lower() == "localfs":
-            return LocalFSReportStore.from_config(config_name)
+        # Convert old configuration parameters to those expected by
+        # DjangoStorageReportStore for backward compatibility
+        config = getattr(settings, config_name, {})
+        storage_type = config.get('STORAGE_TYPE', '').lower()
+        if storage_type == 's3':
+            return DjangoStorageReportStore(
+                storage_class='storages.backends.s3boto.S3BotoStorage',
+                storage_kwargs={
+                    'bucket': config['BUCKET'],
+                    'location': config['ROOT_PATH'],
+                    'querystring_expire': 300,
+                    'gzip': True,
```

Comment:
We've always used gzip because of how large the files get -- in the past I think we just did it explicitly before uploading. So no change in performance I expect.